pyndustri.pyi

```diff
@@ -52,10 +52,118 @@ class Env:
         Underscores become dashes (e.g. `Env.titanium_conveyor` becomes `@titanium-conveyor`).
         """
 
+class World:
+    """
+    World processor specific items
+    """
+
+    @staticmethod
+    def fetch(type: str, team: str, index: int):
+        """Fetches the item at a certain index. Type: {unit|player|core}."""
+    @staticmethod
+    def count(type: str, team: str) -> int:
+        """Get the thing count of a team, where thing: {unit|player|core}."""
+    @staticmethod
+    def fetch_block(type: str, team: str, index: int):
+        """Get the block of type type (eg "Env.conveyor") at index index."""
+    @staticmethod
+    def block_count(type: str, team: str) -> int:
+        """Get the block count of type type (eg "Env.conveyor")."""
+    @staticmethod
+    def get_block(type: str, x: int, y: int):
+        """Get the thing at x, y, type: {floor|ore|block|building}."""
+    @staticmethod
+    def set_block(x: int, y: int, block: str, team: str, rotation: int = 0):
+        """Sets the block at (x, y)."""
+    @staticmethod
+    def set_ore(x: int, y: int, type: str):
+        """Sets the ore at (x, y)."""
+    @staticmethod
+    def set_floor(x: int, y: int, type: str):
+        """Sets the floor at (x, y)."""
+    @staticmethod
+    def spawn_unit(type: str, x: int, y: int, team: str, rot: int):
+        """Spawns a unit at (x, y), then returns it."""
+    @staticmethod
+    def spawn_natural_wave():
+        """Spawns the next wave set by the map at a spawn point i think????"""
+    @staticmethod
+    def spawn_wave(x: int, y: int):
+        """Spawns the next wave set by the map at (x, y) (?)"""
+    @staticmethod
+    def apply_status(unit: Unit, status: str, length: float):
+        """Applys a status to a unit for a certain length"""
+    @staticmethod
+    def clear_status(unit: Unit, status: str):
+        """Clear status from unit"""
+    @staticmethod
+    def set_rate(ipt: int):
+        """Sets the instructions per tick for this processor (note, uses ticks instead of seconds (1 tick = 1/60))"""
+    @staticmethod
+    def notify():
+        """
+        Puts the print buffer on the top of the screen with a fancy panel
+        Use print() without flushing to populate the print buffer
+        """
+    @staticmethod
+    def announce(time: float):
+        """
+        Puts the print buffer in the center of the screen for a certain amount of time
+        """
+    @staticmethod
+    def toast(time: float):
+        """
+        Puts the print buffer slightly lower than the top of the screen with no fancy panel
+        """
+    @staticmethod
+    def mission():
+        """
+        Changes the "mission" from Wave 1... waiting (etc) to the print buffer
+        """
```

Comment:
These are a bit weird. Maybe it's better to have something like:

```python
print('message goes here', file=World.notification)  # or .announcement, or .toast, or .mission
```

Reply:
But then you cant fill the buffer

Reply:
You can still print without flushing and then do a final print to the "file“ you want to use. I think it makes more sense like this, than to have a completely separate method "flush" elsewhere.